src/event.c

```diff
@@ -57,15 +57,30 @@ static void x_check_cb(EV_P_ ev_check *w, int revents) {
             continue;
         }
 
+        bool ignore_event = false;
+
         if (config.ignore_scrolling && cookie->evtype == XI_RawButtonPress) {
             const XIRawEvent *data = (const XIRawEvent *) cookie->data;
             if (data->detail == 4 || data->detail == 5) {
-                XFreeEventData(display, cookie);
-                continue;
+                ignore_event = true;
+            }
+        }
+
+        if (config.ignore_buttons.how_much_symbols
```

Comment:
There's a lot of duplication going on here. We should refactor this a bit to something like this:

```
static boolean is_button_ignored(const XIRawEvent *data) {
    if (config.ignore_scrolling && (data->detail == 4 || data->detail == 5)) {
        return true;
    }

    for (int i = 0; i < config.ignore_buttons.count; i++) {
        if (data -> detail == config.ignore_buttons.buttons[i]) {
            return true;
        }
    }

    return false;
}

// …

if (cookie->evtype == XI_RawButtonPress) {
    const XIRawEvent *data = (const XIRawEvent *) cookie->data;
    if (is_button_ignored(data)) {
        XFreeEventData(display, cookie);
        continue;
    }
}
```